### PR TITLE
Implement the TSV/SVG/PDF download for bar/pie charts 

### PR DESCRIPTION
**Pull request by zhx828**

### CODE REVIEW

File: app/scripts/views/components/pieChart/pieChartTemplate.js [hunk 1]
```diff
@@ -40,7 +40,7 @@
               '@mouseenter="mouseEnter($event)" @mouseleave="mouseLeave($event)">' +
               '<chart-operations :has-chart-title="hasChartTitle" :display-name="displayName" :show-table-icon.sync="showTableIcon" ' +
               ' :show-pie-icon.sync="showPieIcon" :chart-id="chartId" :show-operations="showOperations" :groupid="groupid" ' +
-              ':reset-btn-id="resetBtnId" :chart="chartInst" :attributes="attributes"></chart-operations>' +
+              ':reset-btn-id="resetBtnId" :component="piechart" :chart="chartInst" :attributes="attributes"></chart-operations>' +
```
Comment (kalletlak) [inline app/scripts/views/components/pieChart/pieChartTemplate.js]: 'component' is a predefined word in vue. Instead use something like 'component-type'

File: app/scripts/views/components/chartOperationsHeader.js [hunk 2]
```diff
@@ -53,7 +53,7 @@
             '</div>' +
             '</div>',
     props: [
-      'showOperations', 'resetBtnId', 'chart', 'groupid', 'hasChartTitle', 'showTable', 'displayName', 'chartId', 'showPieIcon', 'showTableIcon','showLogScale','showSurvivalIcon'
+      'showOperations', 'resetBtnId', 'chart', 'component', 'groupid', 'hasChartTitle', 'showTable', 'displayName', 'chartId', 'showPieIcon', 'showTableIcon','showLogScale','showSurvivalIcon'
```
Comment (kalletlak) [inline app/scripts/views/components/chartOperationsHeader.js]: Change 'component' to 'componentType'

File: app/scripts/views/components/chartOperationsHeader.js [hunk 3]
```diff
@@ -146,15 +149,15 @@
...
+          render: function(event, api) {
+            $('#' + chartId + '-pdf', api.elements.tooltip).click(function() {
+              console.log(self.$parent);
             });
             $("#" + chartId + "-svg", api.elements.tooltip).click(function () {
               console.log('download svg')
             });
             $("#" + chartId + "-tsv").click(function () {
-              console.log('download tsv')
+              iViz.util.download(self.component.getChartType(), 'tsv', self.component.getDownloadData('tsv'));
```
Comment (kalletlak) [inline app/scripts/views/components/chartOperationsHeader.js]: Change 'component' to 'componentType'

**Comment by zhx828** [comment]
This relates to #56

File: app/index.html [hunk 4]
```diff
@@ -61,6 +61,8 @@
   <!-- build:js scripts/scripts.js -->
   <script src="https://rawgit.com/notifyjs/notifyjs/master/dist/notify.js"></script>
   <script src="/bower_components/fixed-data-table/dist/fixed-data-table.js"></script>
+  <script src="/app/_components/FileSaver.min.js"></script>
+  <script src="/app/_components/cbio-util.js"></script>
```
Comment (kalletlak) [inline app/index.html:66]: Are these .js libraries same as what present in cbio?

File: app/scripts/views/components/barChart/barChart.js [hunk 5]
```diff
@@ -287,20 +322,26 @@
         }
       }
       chartInst_.render();
+
+      initTsvDownloadData();
+      initCanvasDownloadData();
```
Comment (kalletlak) [inline app/scripts/views/components/barChart/barChart.js:327]: Why to initialize downloadData functions by default after rending the chart? Why don't we do it at the time of click event.

File: app/index.html (hunk 4 above)
Comment (zhx828) [inline app/index.html:66]: Yes they are. I think we should remove them when moving to cbioportal

File: app/scripts/views/components/barChart/barChart.js (hunk 5 above)
Comment (zhx828) [inline app/scripts/views/components/barChart/barChart.js:327]: Good point. I'm thinking to add 'initDownloadData' function in generalChart. But don't a good solution to implement it yet.

**Comment by zhx828** [comment]
#76 #56 are related

**Pull request merged**